### ISSUE TITLE
Check the column/cell visibility

### ISSUE THREAD

**Issue by felicoz** [schrodinger/fixed-data-table-2#650]
I want to do certain actions when an item is visible in viewport, but the column or cell component does not return any visibility value so i have to create my own onViewport hook in each cell, which affects the performance.

## Expected Behavior
To return a visible tag from Column that I can pass to header cell and body cell

## Current Behavior
i see the visible: true/false in the higher component of Column, but we cant pull that value.

## Possible Solution
we can get the visible value from the cell prop or headerCell prop from the Column

## Steps to Reproduce (for bugs)
N/A

## Context
N/A

## Your Environment
N/A

* Version used: 1.1.3
* Browser Name and version: Chrome 103.0.5060.53 
* Operating System and version (desktop or mobile): Mac M1 Monterey
* Link to your project: N/A

**Comment by felicoz** [comment]
hey @pradeepnschrodinger thanks for quick reply!

I have a list of products for sale. We load the price of each individual product separately with another price api. For better performance, I load the price api only when the table item(product) is in the viewport. In my case, I want to be able to pull the visible tag from the Column cell prop, so I can pass the visible tag to my custom cell component. 

thanks!

`
<Column
  cell={({ rowIndex, visible }) => {
    return <MyPriceCell visible={visible />
  }}
/>
`

**Comment by pradeepnschrodinger** [comment]
Thanks for the explanation.

Regarding your usecase, one way to solve it would be to simply tell FDT that it should only render cells inside the visible viewport.

You can set the Table prop `bufferRowCount` to zero, to make sure FDT only **renders visible rows**.
You can then combine this with the Column prop `allowCellsRecycling` to make sure that **only the visible columns** are rendered.

So in this case, if `<MyPiceCell>` is rendered, then you be sure it's always `visible`.
Something like:
```js
<Table ... bufferRowCount={0}>
  <Column
     allowCellsRecycling={true}
     cell={({ rowIndex }) => {
        return <MyPriceCell visible={true} />
     }}
  />
</Table>
```

**Comment by felicoz** [comment]
thanks so much! would definitely give a try

**Comment by pradeepnschrodinger** [comment]
Hey @felicoz, just checking on things since there hasn't been much updates in a while.
Did the snippet above workout for you? Let me know if you need help with anything.

**Comment by felicoz** [comment]
hey @pradeepnschrodinger unfortunately the snippet isn't really working as expected. I really hope we have a way to control/modify the table "row" template, like column and cell.

```
<Table
row={({ rowIndex, children }) => {
  // do something we want on row level
  <div>{children}</div>
}
/>
```

**Comment by pradeepnschrodinger** [comment]
@felicoz , thanks for the update.

Looks like the easiest and proper way to solve this would be to have FDT pass down a `isVisible` prop to the cell component.
This is already implicitly calculated around our [`FixedDataCellGroup`](https://github.com/schrodinger/fixed-data-table-2/blob/v1.2.x/src/FixedDataTableCellGroup.js#L97-L98) component. We might just need to do another calculation to check if the cell is vertically within the viewport bounds, and then pass it down to the cell renderers.

I'll be happy to consider a PR for this.

**Comment by pradeepnschrodinger** [comment]
@felicoz, I explored exposing visibility of cells through a new cell prop. This is outlined in PR (#658).

With this, you should be able to extract out the visibility through something like:
```js
<Table>
  <Column
     cell={({ rowIndex, columnKey, isVisible }) => {
        return <MyPriceCell isVisible={isVisible} />
     }}
  />
</Table>
```

Will this approach work out for your use case?

**Comment by pradeepnschrodinger** [comment]
Sorry for the delays @felicoz.

The proposed changes are now available through [v1.2.13](https://github.com/schrodinger/fixed-data-table-2/releases/tag/v1.2.13), where the cell renderer has a new prop `isVisible`.
This can be used to detect if the current cell is visible (or within the viewport).

I'm closing this since the issue should now be resolved, but feel free to reopen/comment here if you need any help.
